src/core/eval.ml

```diff
@@ -428,3 +428,6 @@ let eq_constr : constr -> constr -> bool = fun (ctx1,t1,u1) (ctx2,t2,u2) ->
   let u2,_ = Ctxt.to_abst ctx2 u2 in
   (eq_modulo [] t1 t2) && (eq_modulo [] u1 u2) ||
   (eq_modulo [] t1 u2) && (eq_modulo [] t2 u1)
+
+let compare_constr : constr -> constr -> int = fun c1 c2 ->
+  if eq_constr c1 c2 then 0 else 1
```

Comment:
Is it possible to just do this? After the OCaml doc, comparison functions must satisfy stronger properties:
Following http://caml.inria.fr/pub/docs/manual-ocaml/libref/List.html:
```
val sort_uniq : ('a -> 'a -> int) -> 'a list -> 'a list
Same as List.sort, but also remove duplicates.
see Array.sort for a complete specification
```
and http://caml.inria.fr/pub/docs/manual-ocaml/libref/Array.html:
```
The following must be true for all x, y, z in a:
    cmp x y > 0 if and only if cmp y x < 0
```
Hence, for sort_uniq to work properly, it seems that we should replace 1 by `Stdlib.compare c1 c2`.
Anyway, even if it works with the current definition, it is kind of dangerous to let such a definition without a comment.
For the record, the code of Stdlib.sort_uniq can be seen on https://github.com/ocaml/ocaml/blob/trunk/stdlib/list.ml .

Reply:
For the moment, I propose you to just add a comment with a WARNING saying that this function does NOT satisfy the requirements mentioned above.

Reply:
fixed (we rely on Stdlib.compare for the non equal case)